server/server.js

```diff
@@ -0,0 +1,264 @@
+        const filePath = join(this.distDir, 'chunks', params.id)
+        return this.serveStatic(req, res, filePath)
+      },
+      // Serves the source maps for specific page bundles
+      '/_next/:buildId/page/:path*.js.map': async (req: IncomingMessage, res: ServerResponse, params: {path: [string]}): Promise<Envelope> => {
+        const paths = params.path || ['']
+        const page = `/${paths.join('/')}`
+        const filePath = join(this.distDir, 'bundles', 'pages', `${page}.js.map`)
+        return this.serveStatic(req, res, filePath)
+      },
+      // Serves the page bundles
+      '/_next/:buildId/page/:path*.js': async (req: IncomingMessage, res: ServerResponse, params: {path: [string]}): Promise<Envelope> => {
+        const paths = params.path || ['']
+        const page = `/${paths.join('/')}`
+        const filePath = join(this.distDir, 'bundles', 'pages', `${page}.js`)
+        return this.serveStatic(req, res, filePath)
+      },
+      // Serves the .next/static files
+      '/_next/static/:path*.js.map': async (req: IncomingMessage, res: ServerResponse, params: {path: [string]}): Promise<Envelope> => {
+        const filePath = join(this.distDir, 'static', ...(params.path || []))
+        return this.serveStatic(req, res, filePath)
+      },
+      // It's very important keep this route's param optional.
+      // (but it should support as many as params, seperated by '/')
+      // Othewise this will lead to a pretty simple DOS attack.
+      // See more: https://github.com/zeit/next.js/issues/2617
+      '/static/:path*': async (req: IncomingMessage, res: ServerResponse, params: {path: [string]}): Promise<Envelope> => {
+        const filePath = join(this.distDir, 'static', ...(params.path || []))
+        return this.serveStatic(req, res, filePath)
+      }
+    }
+
+    if (useFileSystemPublicRoutes) {
+      routes['/:path*'] = async (req: IncomingMessage, res: ServerResponse, params: {path: [string]}, parsedUrl: ParsedUrl) => {
+        const { pathname, query } = parsedUrl
+        return this.render(req, res, pathname, query, parsedUrl)
+      }
+    }
+  }
+
+  async render (req: IncomingMessage, res: ServerResponse, pathname: string, query: {[string]: *}, parsedUrl: ParsedUrl): Promise<Envelope> {
+    return {
+      statusCode: 404,
+      content: STATUS_CODES[404]
+    }
+  }
+
+  async prepare (): Promise<void> {
+    await this.defineRoutes()
+  }
+
+  async handleRequest (req: IncomingMessage, res: ServerResponse, parsedUrl?: UrlParseResult): Promise<void> {
+    const url = normalizeUrl(req, parsedUrl)
+
+    try {
+      await this.run(req, res, url)
+    } catch (err) {
+      this.log(() => console.error(err))
+      res.statusCode = 500
+      res.end(STATUS_CODES[500])
+    }
+  }
+
+  getRequestHandler (): (req: IncomingMessage, res: ServerResponse, parsedUrl?: UrlParseResult) => Promise<void> {
```

Comment:
This doesn't use all args

Reply:
It returns a function 🕵️

Reply:
oh yeah, I read:

Reply:
```

Reply:
```

Reply:
```
getRequestHandler (): (req: Incomi...
```
as
```
getRequestHandler (req: Incomi
```

Reply:
👌 ideally we do `typeof` as we talked about